File: PerfTest2/PerfTest2-Xamarin/PerfTest2-ClassicXamarin.Android/Fragments/DisplayTextFileFragment.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using Android.App;
using Android.Content;
using Android.OS;
using Android.Runtime;
using Android.Views;
using Android.Widget;
using PerfTest2Xamarin.Adapters;

namespace PerfTest2Xamarin.Fragments
{
    public class DisplayTextFileFragment : Fragment
    {
        private string directory;

        public DisplayTextFileFragment(string directory)
            : base()
        {
            this.directory = directory;
        }
    
        public override void OnStart()
        {
            base.OnStart();
            ListView lstFileContents = (ListView)this.Activity.FindViewById(Resource.Id.lstFileContents);
            var adapter = new FileTableAdapter(this.Activity, directory);
            lstFileContents.Adapter = adapter;

        }

        public override View OnCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState)
        {
            return inflater.Inflate(Resource.Layout.fragment_display_text_file, container, false);
        }
    }
}

File: PerfTest2/PerfTest2-Java/app/src/main/java/com/vandammeford/kevinf/perftest2_java/DisplayTextFileFragment.java
package com.vandammeford.kevinf.perftest2_java;

import android.app.Activity;
import android.app.AlertDialog;
import android.net.Uri;
import android.os.Bundle;
import android.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ListView;
import android.widget.TextView;

public class DisplayTextFileFragment extends Fragment {

    private OnFragmentInteractionListener mListener;

    public static DisplayTextFileFragment newInstance() {
        DisplayTextFileFragment fragment = new DisplayTextFileFragment();
        Bundle args = new Bundle();
        fragment.setArguments(args);
        return fragment;
    }

    public DisplayTextFileFragment() {
        // Required empty public constructor
    }

    @Override
    public void onStart() {
        super.onStart();

        ListView lstFileContents = (ListView)this.getActivity().findViewById(R.id.lstFileContents);
        FileTableAdapter adapter = new FileTableAdapter(this.getActivity());
        lstFileContents.setAdapter(adapter);
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_display_text_file, container, false);
    }

    // TODO: Rename method, update argument and hook method into UI event
    public void onButtonPressed(Uri uri) {
        if (mListener != null) {
            mListener.onFragmentInteraction(uri);
        }
    }

    @Override
    public void onAttach(Activity activity) {
        super.onAttach(activity);
        try {
            mListener = (OnFragmentInteractionListener) activity;
        } catch (ClassCastException e) {
            throw new ClassCastException(activity.toString()
                    + " must implement OnFragmentInteractionListener");
        }
    }

    @Override
    public void onDetach() {
        super.onDetach();
        mListener = null;
    }

    public interface OnFragmentInteractionListener {
        // TODO: Update argument type and name
        public void onFragmentInteraction(Uri uri);
    }

}
